### PR TITLE
Centralize mock test data

### PR DESCRIPTION
**Pull request by sharrief**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Tests were broken and using different fake data. This change centralizes all the test data into one file and adds a mock module for DataService so individual tests dont need to mock the DataService unless they have specific needs (like needing to force a DataService call to throw an error)

- Removed recent tickets (it was for sprint-3 demo). Replaced it with All tickets hook (which will be used by dashboard lists in another PR).
- Removed hardcoded test data in test files and replaced it with references to the centralized test data.
- Added mock module for DataService which uses the centralized test data.
- Updated ticket schema to include `modified_at` which will be used for dashboard lists (like recently closed).

## Related Tickets & Documents

N/A

## QA Instructions, Screenshots, Recordings

N/A

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [ ] Semantic HTML implemented?
- [ ] Keyboard operability supported?
- [ ] Color contrast tested?

## Added/updated tests?
_We are aiming for 100% statement coverage._

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

### Swizzle
swizzle

### CODE REVIEW

File: package.json
[the remark this answers is not in the document]
Comment (DillonOLeary) [reply package.json:1]: The other issue is that these commands would likely run in the dev container. That's one of the issues I ran into when trying to set up the database locally. I don't know if that means we shouldn't have them, but we would at least need to include some kind of note that the commands need to be run locally

File: __tests__/src/services/DataService.test.ts [hunk 1]
```diff
@@ -131,26 +133,32 @@ describe('DataService', () => {
...
-      }))
+      }));
       mockSupabaseClient.setTestData(expectedQueryResults);
       const expectedServiceRequestSummary = expectedQueryResults.map((data) => {
         const {
           clients, pets, team_members, id, label, ...ticket
         } = data;
         return {
-          id: id,
+          id,
```
Comment (DillonOLeary) [inline __tests__/src/services/DataService.test.ts:147]: nit - is this change needed?

File: src/components/TicketsTable.tsx
Comment (DillonOLeary) [inline src/components/TicketsTable.tsx:1]: This is unrelated to the test data change right? I assume this is just a typescript change that doesn't impact functionality

File: src/hooks/__mocks__/useTicketById.ts
Comment (DillonOLeary) [inline src/hooks/__mocks__/useTicketById.ts:1]: Good change here - really cleans up the code!

File: src/hooks/useAllTickets.ts [hunk 2]
```diff
@@ -1,19 +1,16 @@
 import { ServiceRequestSummary } from '@types';
-import { useSearchParams } from 'next/navigation';
 import { useEffect, useState } from 'react';
 import * as DataService from '@services/DataService';
 
-export default function useRecentTickets() {
+export default function useAllTickets() {
   const [tickets, setTickets] = useState<(ServiceRequestSummary)[]>([]);
-  // Hack:  using url change to trigger to get recent tickets
-  const params = useSearchParams();
```
Comment (DillonOLeary) [inline src/hooks/useAllTickets.ts:9]: What is the impact of getting rid of this code? Does it mean that the recent tickets table is no longer being updated when a new ticket is added?

File: src/services/DataService.ts [hunk 3]
```diff
@@ -96,11 +96,13 @@ export async function getServiceRequestSummary(): Promise<ServiceRequestSummary[
...
     pets(name),
-    team_members(first_name),
-    urgent
+    team_members(first_name, email),
+    urgent,
+    status,
+    modified_at
     `)
-    .order('created_at', { ascending: false })
-    .limit(10);
```
Comment (DillonOLeary) [inline src/services/DataService.ts:103]: I know we talked about how the browser can store a lot more than 10 rows so this is a good change. I wonder though if we should still have an upper limit, like 100,000. For instance, what if there is a bug and millions of rows are added, it could be a major problem for the client

**Review by DillonOLeary — COMMENTED** [review]
Smart change here! This should make testing a lot easier going forward.

It seems like there are some other small changes baked into this PR as well, like the `modified_at` change and the package json change. I know there's overhead to splitting the PR up, so I'm not too concerned unless one of those changes sparks debate which blocks unrelated code from merging.

**Comment by AdelBeit** [comment]
@sharrief @DillonOLeary I'm a little disoriented from the break i took to move, i was going to refactor my tests for #74 but I think i need to wait till this PR is merged so that i could rebase on top of the new testing method, am I correct that this PR needs to be merged first?

File: utils/TestData.ts
[the remark this answers is not in the document]
Comment (sharrief) [reply utils/TestData.ts:1]: Adding this file to that folder causes the full test run to fail bc jest assumes every file in __tests__ is a test with runnable test code, and this file doesn't contain that.

File: package.json
[the remark this answers is not in the document]
Comment (sharrief) [reply package.json:1]: 1. Didn't know about yarn exec, will see if that works we well!

2. Yes, these commands need to be run in a local terminal, but we can't aadd comments to a json file, meaning the README is the best place to put that info. Will update PR with README modification

File: __tests__/src/services/DataService.test.ts (hunk 1 above)
Comment (sharrief) [reply __tests__/src/services/DataService.test.ts:147]: I think linting fixed this, but in general this is good style.

File: src/hooks/useAllTickets.ts (hunk 2 above)
Comment (sharrief) [reply src/hooks/useAllTickets.ts:9]: Good catch, this hack to trigger a refresh needs to be replaced with a SWR mutate call inside of the dialog that creates the tickets. Will update that in a new commit.

File: src/components/TicketsTable.tsx
Comment (sharrief) [reply src/components/TicketsTable.tsx:1]: No this is a functional change. The team_member prop was updated in the supabase query from being just the teammate's first name to being an object with their first name and their email.

This PR makes some schema changes that are used by the tickets table PR. I made the changes in this PR bc the changes also require updates to the mock test data and I want to use this mock test data setup to test the all tickets table. Since this isn't merged yet, if I added all these faker changes to the ticket table PR it would have bloated that PR considerably with duplicated code. (stacked PRs would've been helpful here, but wasn't aware at the time)

File: package.json
[the remark this answers is not in the document]
Comment (sharrief) [reply package.json:1]: I updated the README, but don't want to update the command to use yarn exec because this command runs on the local PC, and the local PC might not have yarn, but npx comes bundled with npm and node.

I intentionally minimize the amount of global commands I have on my host machine and do most dev in containers. Running npx here instead of yarn exec won't have any affect on our yarn.lock or anything else.

File: src/hooks/useAllTickets.ts (hunk 2 above)
Comment (sharrief) [reply src/hooks/useAllTickets.ts:9]: Whoops nevermind, its the other PR that has SWR, so I'll revert this change and fix in the other PR

**Comment by sharrief** [comment]
Don't merge this until the PR for the tables is ready, as that PR is merged into this branch

File: utils/TestData.ts
[the remark this answers is not in the document]
Comment (AdelBeit) [reply utils/TestData.ts:1]: I think this is fine under utils/, if we really want to we can add an empty test inside testData.ts just so jest wont yell at us

File: src/components/TicketsTable.tsx [hunk 4]
```diff
@@ -28,6 +30,14 @@ function CreatedAtTemplate({ created_at, id }) {
   );
 }
 
+function TeamMemberView({ id, team_member }: ServiceRequestSummary) {
+  return (
+    <Link className="text-gray-900" href={`?ticket=${id}`} key={id}>
```
Comment (AdelBeit) [inline src/components/TicketsTable.tsx:35]: I thought we did away with <links> and had a single onclick for each row, is this code just outdated?

File: src/hooks/__mocks__/useTeamMembers.ts
Comment (AdelBeit) [inline src/hooks/__mocks__/useTeamMembers.ts:1]: now that we have centralized the testdata in utils/testdata.ts should we remove the entire mocks folder? or at least replace the mockteammebers object to be the one imported from utils/testdata?

File: src/services/__mocks__/DataService.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
+// TODO remove after moving to prettier
```
Comment (AdelBeit) [inline src/services/__mocks__/DataService.ts:1]: is this comment still relevant?

File: src/hooks/__mocks__/useTeamMembers.ts
Comment (sharrief) [reply src/hooks/__mocks__/useTeamMembers.ts:1]: You're right, thanks for the catch!

File: src/components/TicketsTable.tsx (hunk 4 above)
Comment (sharrief) [reply src/components/TicketsTable.tsx:35]: I probably brought this back in while merging. It was a complex merge and I missed this, and maybe more. Will fix it

File: src/services/__mocks__/DataService.ts (hunk 5 above)
Comment (sharrief) [reply src/services/__mocks__/DataService.ts:1]: There was probably a eslint disable comment here previously. I'l clean this up

**Comment by sharrief** [comment]
@AdelBeit, you left some great feedback and caught some good things I missed in the code review. It’s awesome to see that you understand the codebase so well and were able to navigate a dense PR. Kudos and thanks for the thorough review!

**Review by AdelBeit — APPROVED** [review]
this is a very significant refactor thank you so much for doing it!

**Comment by AdelBeit** [comment]
> @AdelBeit, you left some great feedback and caught some good things I missed in the code review. It’s awesome to see that you understand the codebase so well and were able to navigate a dense PR. Kudos and thanks for the thorough review!

You're very welcome! I appreciate your kind words 🙂
